### ISSUE TITLE
Markdown links to Markdown pages with explicit page command are broken

### ISSUE THREAD

**Issue by arwedus** [doxygen/doxygen#8528]
Hi @albert-github , I have now created a separate task for the bug in #8378, I hope this can be fixed quickly... I didn't find a quick solution.

**Bug description**

If you start a markdown page with a `\page` command, Markdown links to the page lead to a non-existing page instead.

See also: #8378 

Example page:

```
\page p_input_handling Input Handling

# Input Handling

...
```

Example page 2:

```
subpages:
- \subpage p_input_handling input handling

Please refer to [input handling](doc/input_handling.md) documentation for details.
```

If a markdown page starts with just the L1 heading, links in Markdown style like `[output handling](doc/output_handling.md)` lead to the right page.

**Expected behavior:**

- Generated HTML for a relative markdown link to another markdown page works as expected and opens the page.

**To Reproduce:**

See attached example (much reduced version of #8378).

**Version:**

Tested on Doxygen 1.9.1, but present in older versions too.

[example_broken_md_link.tar.gz](https://github.com/doxygen/doxygen/files/6406309/example_broken_md_link.tar.gz)

**Comment by albert-github** [comment]
I reduced the small problem even a little bit further to:

doc/input_handling.md
```
@page input-handling Input Handling

# Input Handling

Data is further processed by the [output handling](output_handling.md).
```

and doc/output_handling.md
```
# Output Handling

- runs after [input handling](input_handling.md) (broken link)
- format output data after triggered by \ref input-handling (this link works)
```

To understand what happen we have to run doxygen so it outputs as well the result of the internal markdown conversion (so `doxygen -d markdown`):

For input_handling.md this is:
```
======== Markdown =========
---- input -------
@page input-handling Input Handling

# Input Handling

Data is further processed by the [output handling](output_handling.md).

---- output -----
@page input-handling Input Handling

@section autotoc_md0 Input Handling

Data is further processed by the @ref .../issue_8528/example_broken_md_link/doc/output_handling.md "output handling".

=========
```
and for output_handling.md:
```
======== Markdown =========
---- input -------
@page md_doc_output_handling Output Handling\ilinebr

- runs after [input handling](input_handling.md) (broken link)
- format output data after triggered by \ref input-handling (this link works)

---- output -----
@page md_doc_output_handling Output Handling\ilinebr

- runs after @ref .../issue_8528/example_broken_md_link/doc/input_handling.md "input handling" (broken link)
- format output data after triggered by \ref input-handling (this link works)

=========
```

One thing that immediately catches the eye is that the input for output_handling.md contains the, automatic, link like:
```
@page md_doc_output_handling Output Handling\ilinebr
```
and later on doxygen will use this link to be able to link to the file output_handling.md (so both are internal links as discussed in https://github.com/doxygen/doxygen/issues/8378#issuecomment-830220454).
Maybe we have to think about the possibility to generate in case of `@page` also the label `md_...` by means of an anchor so we can refer to it (first tests look promising, but more tests have to be run).

Note: the fact that:
```
- \subpage md_doc_output_handling
```
is working is a little bit of luck as here you use a generated label, better would be to use an explicit label. This type of usage is discouraged.

**Comment by arwedus** [comment]
not sure what `\ilinebr` is, but this is something that's added in `parseInput`, except if if `isExplicitPage()` - so if that is required, than implementing something for the condition `isExplicitPage` might fix it.

p.s.: I know that using the generated labels is not an intended way, that's why I'm making use of the `\page` command at the top ;-).

**Comment by arwedus** [comment]
I gave it a shot... https://github.com/arwedus/doxygen/tree/feature/issue-8378-implicit-md-anchor makes the document with the explicit `\page` command look just like the one without explicit page command (except that I get to choose my own page name).

Bad thing is, this did not fix the broken link on "input_handling.md" in the HTML output :(

**Comment by albert-github** [comment]
The `\ilinebr` is the, internal, coding for a newline. This command is automatically added( and later on parsed) and is necessary to keep in the next stage the line count correct for e.g. warnings and line references. When not using `\ilinebr` but a regular newline the line count would have been increased and thus the line number / line reference would be wrong.

**Comment by arwedus** [comment]
@albert-github any updates on this? I saw that you merged a PR a while ago that improves debug output for Markdown procesor, but did you also find a cure for the problem with the broken page links?

Oh and btw., it's been more than half a year since the last Doxygen minor release, can you say something about when you'll create release 1.9.2, @doxygen ? 🙂

**Comment by doxygen** [comment]
@arwedus I plan to release a new version somewhere in August (during my holidays).

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8896

**Comment by albert-github** [comment]
Slightly adjusted code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

@arwedus please test.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
